app/vmselect/promql/rollup.go

```diff
@@ -900,17 +900,26 @@ func getMaxPrevInterval(scrapeInterval int64) int64 {
 	return scrapeInterval + scrapeInterval/8
 }
 
-func removeCounterResets(values []float64) {
+func removeCounterResets(values []float64, timestamps []int64, maxStalenessInterval int64) {
 	// There is no need in handling NaNs here, since they are impossible
 	// on values from vmstorage.
 	if len(values) == 0 {
 		return
 	}
 	var correction float64
 	prevValue := values[0]
+	prevTS := timestamps[0]
 	for i, v := range values {
 		d := v - prevValue
 		if d < 0 {
+			currTS := timestamps[i]
```

Comment:
It is better from performance PoV to move this line inside the `if maxStalenessInterval > 0 { ... }` block. In this case timestamps will be loaded only if this condition is met, while in default case (when `maxStalenessInterval=0`) the performance will remain the same as before adding this change.